src/mpid/ch4/shm/posix/posix_init.c

```diff
@@ -148,6 +161,61 @@ static int init_vci(int vci)
     goto fn_exit;
 }
 
+static void init_topo_info(void)
+{
+    hwloc_topology_t topology;
+    hwloc_cpuset_t cpuset;
+    MPIDI_POSIX_topo_info_t *topo = &MPIDI_POSIX_global.topo;
+
+    hwloc_topology_init(&topology);
+    hwloc_topology_load(topology);
```

Comment:
can we use the `MPIR_hwtopo` API instead? another hwloc init could cause startup performance issues.

Reply:
Sure. I can do that.